sky/core.py

```diff
@@ -551,10 +560,22 @@ def cancel(cluster_name: str,
         cluster_name, operation_str='Cancelling jobs')
 
     # Check the status of the cluster.
-    handle = backend_utils.check_cluster_available(
-        cluster_name,
-        operation='cancelling jobs',
-    )
+    try:
+        handle = backend_utils.check_cluster_available(
+            cluster_name,
+            operation='cancelling jobs',
+        )
+    except exceptions.ClusterNotUpError as e:
+        if not _ignore_cluster_aliveness:
+            raise
+        if (not isinstance(e.handle, backends.CloudVmRayResourceHandle) or
```

Comment:
From the docstr of check_cluster_available() it looks like
```
        exceptions.NotSupportedError: if the cluster is not based on
          CloudVmRayBackend.
```
will be thrown, and this check here is not needed?